File: app/src/main/java/com/magnitudestudios/GameFace/Interfaces/RoomCallback.java
package com.magnitudestudios.GameFace.Interfaces;

import com.magnitudestudios.GameFace.pojo.IceCandidatePOJO;
import com.magnitudestudios.GameFace.pojo.SessionInfoPOJO;

import org.webrtc.IceCandidate;
import org.webrtc.SessionDescription;

public interface RoomCallback {
    void offerReceived(SessionInfoPOJO session);
    void answerReceived(SessionInfoPOJO session);
    void newParticipantJoined(String s);
    void iceServerReceived(IceCandidatePOJO iceCandidate);
    void participantLeft(String s);
    void onJoinedRoom(boolean b);
    void onCreateRoom();
    void onLeftRoom();

}

File: app/src/main/java/com/magnitudestudios/GameFace/Interfaces/RoomCallback.kt
package com.magnitudestudios.GameFace.Interfaces

import com.magnitudestudios.GameFace.pojo.IceCandidatePOJO
import com.magnitudestudios.GameFace.pojo.SessionInfoPOJO

interface RoomCallback {
    fun offerReceived(session: SessionInfoPOJO?)
    fun answerReceived(session: SessionInfoPOJO?)
    fun newParticipantJoined(s: String?)
    fun iceServerReceived(iceCandidate: IceCandidatePOJO?)
    fun participantLeft(s: String?)
    fun onJoinedRoom(b: Boolean)
    fun onCreateRoom()
    fun onLeftRoom()
}